Add `rule-nesting-at-rule-required-list` (#8680)
stylelint/stylelint: lib/rules/declaration-block-nesting-at-rule-required-list/README.md

# declaration-block-nesting-at-rule-required-list

> Require specific at-rules to be present around CSS declarations.

Reviewer: suggestion

Instead of _to be present around_ maybe _to be surrounding_.
Or maybe
```suggestion
Require declarations to be enclosed in specific at-rules.
```
?

<!-- prettier-ignore -->
```css
a { color: red; }
/**    ↑
 * This declaration */
```

> ## Options
> 
> `array|string|regex`: `["layer"]`

Reviewer: see
https://stylelint.io/user-guide/rules/at-rule-property-required-list#options
https://github.com/stylelint/stylelint/pull/8557#discussion_r2072306765

```suggestion
## Options

`Array<string | RegExp>`
```

Given:

```json

> ["layer"]

Reviewer: ```suggestion
{
  "declaration-block-nesting-at-rule-required-list": ["layer"]
}
```
